pandas/io/formats/info.py

```diff
@@ -288,7 +288,11 @@ def _verbose_repr(
         len_column = len(pprint_thing(column_head))
         space = max(max_col, len_column) + col_space
 
-        max_id = len(pprint_thing(col_count))
+        # GH #36765
+        # add one space in max_id because there is a one-space padding
+        # in front of the number
+        # this allows maintain two spaces gap between columns
+        max_id = len(pprint_thing(col_count)) + 1
```

Comment:
Note that this was starting playing role only when number of columns was over 1000. Because below 1000 ``max_id < len_id``. Therefore below 1000 columns there were no issues.